app/models/spree/order_decorator.rb

```diff
@@ -333,6 +333,18 @@ def pending_payments
     payments.select {|p| p.state == "checkout"} # Original definition
   end
 
+  # Although Spree 2 supports multi shipments per order, in OFN we keep the rule one shipment per order
+  #   Thus, this method sets the given shipping_method on the first and only shipment in the order
+  def shipping_method=(shipping_method = nil)
+    if shipments.empty?
+      shipment = Spree::Shipment.new
+      shipment.order = self
+      shipment.save
```

Comment:
AFAIK we don't need this save as AR persists the order and all its associations when doing a `order.save`. Correct me if I'm wrong.